File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  if (nums.length === 0) return 0;
  const set1 = new Set(nums);
  const testArr = Array.from(set1);
  let maxCount = 0;
  for (let i = 0; i < testArr.length; i++) {
    let start;
    let seqLength = 0;
    if (!set1.has(testArr[i] - 1)) {
      start = testArr[i];
      while (set1.has(start)) {
        seqLength++;
        start++;
      }
    }
    maxCount = Math.max(seqLength, maxCount);
  }
  return maxCount;
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
          if (nums.length == 0) {
      return 0;
    }
    Set<Integer> set1 = new HashSet<>();
    for (int i = 0; i < nums.length; i++) {
      set1.add(nums[i]);
    }
    int maxCount = 0;
    for (int i = 0; i < nums.length; i++) {
      int start;
      int seqLength = 0;
      if (!set1.contains(nums[i] - 1)) {
        start = nums[i];
        while (set1.contains(start)) {
          start++;
          seqLength++;
        }
      }
      maxCount = Math.max(maxCount, seqLength);
    }
    return maxCount;
    }
}